File: rs2-rent-sistem.models/EquipmentListItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace rs2_rent_sistem.models
{
    public class EquipmentListItem
    {
        public int Id {  get; set; }
        public string? ImageUrl { get; set; }
        public string ItemName { get; set; }
        public double Rating { get; set; }
        public int NumberOfReviews { get; set; }
        public string FormattedCostPerDay { get; set; }
        public bool IsInCart { get; set; }
    }
}

File: rs2_rent_sistem/lib/models/equipment_list_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'equipment_list_item.freezed.dart';
part 'equipment_list_item.g.dart';

@Freezed()
class EquipmentListItem with _$EquipmentListItem {
  const factory EquipmentListItem({
    @Default('') String id,
    @Default('') String imageUrl,
    @Default('') String itemName,
    @Default('') String manufacturer,
    @Default(0.0) double rating,
    @Default(0) int numberOfReviews,
    @Default('') String costPerDay,
    @Default(false) bool isInCart,
  }) = _EquipmentListItem;

  factory EquipmentListItem.fromJson(Map<String, dynamic> json) => _$EquipmentListItemFromJson(json);
}
